### PR TITLE
Documentation on IRC bridge mode handling

### LINKED ISSUE #1242: Documentation on bridge behaviour with Freenode channel modes
It would be great to have a documentation that shows how bridge behaves with Freenode channel modes https://freenode.net/kb/answer/channelmodes

### PR DESCRIPTION
**Pull request by Half-Shot**
Fixes #1242

### CODE REVIEW

File: docs/irc_modes.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+By default the bridge maps users like so
+
+```yaml
+# taken from config.sample.yaml
+modePowerMap:
+    o: 50
+    v: 1
+```
+
+An operator on the IRC side of the bridge will be given moderator priviledges (PL50) and a voiced user will be given
```
Comment (jaller94) [inline docs/irc_modes.md:24]:
```suggestion
An operator on the IRC side of the bridge will be given moderator privileges (PL50) and a voiced user will be given
```

File: docs/irc_modes.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+## Key protected (+k)
+
+If a channel is protected by a key, it cannot be entered by joining via an alias. Instead you may join by using
+the [`!join`](admin_room.md#join) command. Be aware that the bridge purposefully does not store channel keys in
+the database as a security precaution so you should be expected to do this again on bridge restart.
+
+## Channel Forwarding (+f)
+
+Channel forwarding is presently unsupported, see https://github.com/matrix-org/matrix-appservice-irc/issues/214
+for information and updates
```
Comment (jaller94) [inline docs/irc_modes.md:63]:
```suggestion
for information and updates.
```

File: docs/irc_modes.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+For example to allow matrix.org bridge users to access your invite-only channel, you would do:
+
+```
+/mode #the-secret-cookies +I !@gateway/shell/matrix.org/*
+```
+
+## +s (Secret)
+
+By default the IRC bridge will automatically insert any newly joined rooms into the homeserver's room directory, if
+the room has not been bridged. The +s mode will mark the channel as secret and the bridge will not show it in the
```
Comment (jaller94) [inline docs/irc_modes.md:51]: This condition "not been bridged" contradicts the former statement of "newly joined".
If you meant "not been bridged before", it's already fully explained in the first sentence.

```suggestion
By default the IRC bridge will automatically insert any newly joined rooms into the homeserver's room directory.
The +s mode will mark the channel as secret and the bridge will not show it in the
```

**Review by jaller94 — APPROVED** [review]
Nice article. :+1:

File: docs/irc_modes.md (hunk 3 above)
Comment (Half-Shot) [reply docs/irc_modes.md:51]: That's much better, ty.

**Pull request merged**